idseq_pipeline/commands/host_filtering_functions.py

```diff
@@ -205,64 +208,72 @@ def run_star(fastq_file_1, fastq_file_2):
-        execute_command("cp %s/%s %s/%s;" % (tmp_result_dir, 'Unmapped.out.mate2', RESULT_DIR, STAR_OUT2))
+        if len(fastq_files) == 2:
+            execute_command("cp %s/%s %s/%s;" % (tmp_result_dir, 'Unmapped.out.mate2', RESULT_DIR, STAR_OUT2))
     else:
-        run_star_part(SCRATCH_DIR, REF_DIR + '/STAR_genome', fastq_file_1, fastq_file_2)
+        run_star_part(SCRATCH_DIR, REF_DIR + '/STAR_genome', fastq_files)
         # extract out unmapped files
         execute_command("cp %s/%s %s/%s;" % (SCRATCH_DIR, 'Unmapped.out.mate1', RESULT_DIR, STAR_OUT1))
-        execute_command("cp %s/%s %s/%s;" % (SCRATCH_DIR, 'Unmapped.out.mate2', RESULT_DIR, STAR_OUT2))
+        if len(fastq_files) == 2:
+            execute_command("cp %s/%s %s/%s;" % (SCRATCH_DIR, 'Unmapped.out.mate2', RESULT_DIR, STAR_OUT2))
     # copy back to aws
     execute_command("aws s3 cp %s/%s %s/;" % (RESULT_DIR, STAR_OUT1, SAMPLE_S3_OUTPUT_PATH))
-    execute_command("aws s3 cp %s/%s %s/;" % (RESULT_DIR, STAR_OUT2, SAMPLE_S3_OUTPUT_PATH))
+    if len(fastq_files) == 2:
+        execute_command("aws s3 cp %s/%s %s/;" % (RESULT_DIR, STAR_OUT2, SAMPLE_S3_OUTPUT_PATH))
     # cleanup
     execute_command("cd %s; rm -rf *" % SCRATCH_DIR)
     write_to_log("finished job")
 
-def run_priceseqfilter(input_fq_1, input_fq_2):
+def run_priceseqfilter(input_fqs):
     priceseq_params = [PRICESEQ_FILTER,
                        '-a','12',
-                       '-fp',input_fq_1 , input_fq_2,
-                       '-op',
-                       RESULT_DIR +'/' + PRICESEQFILTER_OUT1,
-                       RESULT_DIR +'/' + PRICESEQFILTER_OUT2,
                        '-rnf','90',
                        '-log','c']
+    if len(input_fqs) == 2:
+        priceseq_params.extend(['-fp', input_fqs[0], input_fqs[1],
+                                '-op', RESULT_DIR + '/' + PRICESEQFILTER_OUT1, RESULT_DIR + '/' + PRICESEQFILTER_OUT2])
+    else:
+        priceseq_params.extend(['-f', input_fqs[0],
+                                '-o', RESULT_DIR + '/' + PRICESEQFILTER_OUT1])
     if "fastq" in FILE_TYPE:
         priceseq_params.extend(['-rqf','85','0.98'])
     execute_command_realtime_stdout(" ".join(priceseq_params))
     write_to_log("finished job")
     # copy back to aws
     execute_command("aws s3 cp %s/%s %s/;" % (RESULT_DIR, PRICESEQFILTER_OUT1, SAMPLE_S3_OUTPUT_PATH))
-    execute_command("aws s3 cp %s/%s %s/;" % (RESULT_DIR, PRICESEQFILTER_OUT2, SAMPLE_S3_OUTPUT_PATH))
+    if len(input_fqs) == 2:
+        execute_command("aws s3 cp %s/%s %s/;" % (RESULT_DIR, PRICESEQFILTER_OUT2, SAMPLE_S3_OUTPUT_PATH))
 
-def run_fq2fa(input_fq_1, input_fq_2):
-    execute_command("sed -n '1~4s/^@/>/p;2~4p' <%s >%s/%s" % (input_fq_1, RESULT_DIR, FQ2FA_OUT1))
-    execute_command("sed -n '1~4s/^@/>/p;2~4p' <%s >%s/%s" % (input_fq_2, RESULT_DIR, FQ2FA_OUT2))
+def run_fq2fa(input_fqs):
```

Comment:
I believe  `input_fq_1` is not defined here.

Reply:
Thanks!